scipy/stats/_survival.py

```diff
@@ -94,7 +127,11 @@ def confidence_interval(self, confidence_level=0.95, *, method='linear'):
         if np.any(np.isnan(low) | np.isnan(high)):
             warnings.warn(message, RuntimeWarning, stacklevel=2)
 
-        return ConfidenceInterval(np.clip(low, 0, 1), np.clip(high, 0, 1))
+        low = EmpiricalDistributionFunction(self.q, np.clip(low, 0, 1),
+                                            None, None, self._kind)
+        high = EmpiricalDistributionFunction(self.q, np.clip(high, 0, 1),
+                                            None, None, self._kind)
+        return ConfidenceInterval(low, high)
```

Comment:
I am not sure I understand why you would want to have the nesting. Why do you want the output of CI to have a full object?

Reply:
See plotting code above, for example. It makes it nicer to be able to evaluate the CI bounds at arbitrary points. Without this, the code might look like:

```python3
# Also note that confidence interval bounds now have `evaluate` method
q = [-3] + list(res.sf.q) + [3]
low, high = res.sf.confidence_interval()
plt.step(q, res.sf.evaluate(q), color='C0', where='post')
plt.step(q, [1, list(low), low[-1]], color='C1', where='post')
plt.step(q, [1, list(high), high[-1]], color='C1', where='post')
```
which just isn't quite as nice.

In general, I don't see why the need to evaluate the CI bounds at arbitrary points is much less than the need to evaluate the point estimate at arbitrary points. So if there needs to be a way to evaluate the point estimate of `sf` at arbitrary points (and this was requested by a few reviewers of gh-18035), why shouldn't its upper and lower bounds?

---

Likewise, I think a very simple plotting method would be useful for the point estimate because it saves the user from needing to remember and execute several steps:
- Evaluate the function not only at all the unique times, but also at a time before the first sample time and after the last sample time. Those edge points are important to make the plot look right.
- Use `plt.step` instead of `plt.plot`
- Set appropriate options (e.g. `where='post'`).

The need to plot the CI bounds is just as great as the need to plot the point estimate, so if the point estimate should get a `plot` method, the CI bounds should get a `plot` method.

Reply:
I would prefer if instead we would allow `res.[sf,cdf].confidence_interval` to get as input the new quantiles: `confidence_interval(confidence_level, *, ..., quantiles)`. This looks natural to me and close to `evaluate`. Then we don't need the object nesting and in the plot function, we can just have an option to plot CI or not. Here it would bring a bit more "legitimacy" to add the `plot` function.

(About `plot`. In general I am not against that, we need to be careful about not doing too much though.)